drivers/hmis/app/graphql/mutations/update_form_definition_typed.rb

```diff
@@ -0,0 +1,42 @@
+#  Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+#  License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+#
+
+module Mutations
+  # Similar to UpdateFormDefinition, but accepts a typed definition instead of a JSON object.
+  # This allows the frontend to be completely shielded from creating and manipulating JSON objects with snake_case keys.
+  class UpdateFormDefinitionTyped < CleanBaseMutation
+    argument :id, ID, required: true
+    argument :definition, 'Types::Admin::FormDefinitionTypedInput', required: false
+    # TODO(#6005) - add role and title as inputs
+    # argument :role, Types::Forms::Enums::FormRole, required: false
+    # argument :title, String, required: false
+
+    field :form_definition, Types::Forms::FormDefinition, null: true
+
+    def resolve(id:, definition:)
+      raise 'not allowed' unless current_user.can_manage_forms?
+
+      form_definition = Hmis::Form::Definition.find_by(id: id)
+      raise 'not found' unless form_definition
+
+      form_definition.definition = definition
```

Comment:
Huh, I'm surprised that this works without calling `definition.to_params` to convert it from a `GraphQL::Schema::InputObject` into a ruby hash. I'm unsure how this behaves (is it documented?) so it might be safe to convert it to params first. (If we stick with this approach)